python/pyspark/worker.py

```diff
@@ -45,6 +45,8 @@
 
 if sys.version >= '3':
     basestring = str
+else:
+    from itertools import imap as map  # use iterator map by default
```

Comment:
It deserves a comment, at least. I think this is a relatively safe change as this is already how Python 3 works.
I can only find one usage in this file, which is the part that applies a UDF to data. That's the source of the issue? just checking. Surprisingly that seems to be the only call in non-test code where it seems to matter.

Reply:
Yes, the only call is the source the trigger the issue.

Reply:
is this not going to potentially change the behavior for python UDF?

Reply:
Hm, that's a good point. I don't know enough Python to be sure. @holdenk do you know how Python would work in this regard? Is it safer to push this check to the one site below and call one or the other map function? it looks like itertools doesn't have imap in Python 3.

Reply:
Are we worrying about the case that global `map` inside the pickled function is overridden by existing global `imap`? That's not going to happen per `https://github.com/cloudpipe/cloudpickle/pull/240`.

Reply:
Is it possible we add a test to verify it in this PR too?

Reply:
Yea, sounds good to have a test.

Reply:
How can we test it here -- make a UDF that checks the value of `map.__module__`? if it's "itertools", then fail, as it would mean this import 'leaked' into the UDF right? Otherwise in Python 2/3 it should return "builtins" or "__builtin__"

Reply:
Yes, I think we can test like that.

Reply:
The way we pickle UDFs is a little weird, so I wouldn't be too surprised if we did end up doing something silly by accident here, in that case we can also invert the imports (e.g. import map as imap in py3)